### ISSUE TITLE
BlockIgnores do not work in .html files

### ISSUE THREAD

**Issue by phxnsharp** [jdkato/txtlint#507]
A BlockIgnores statement does not work for .html files. Hugo allows us to have custom content using a special tag which we would like to have Vale ignore. 

```html
<h2>
Header content
</h2>

	{{< cardpane >}}
	{{< card 
		header="Card Title" >}}
		Content of card
	</div>
	{{< /card >}}
```

Relevant part of .vale.ini
```ini
...
[*.{md,rst,html}]

# Accept HUGO shortcuts/commands
BlockIgnores = (?s) *({{< [^>]* >}})

...
```

The BlockIgnores as specified above does work in .md files.

Since the code to implement BlockIgnores is in the same function as ignoring the front matter, this issue is probably related to #506.

vale version 2.20.2
Ubuntu 20.04.5 LTS

**Comment by jdkato** [comment]
This is expected, [as stated in the docs](https://vale.sh/docs/topics/scoping/#non-standard-markup):

> When working with **non-HTML markup**, you’ll probably find that there are certain non-standard sections of text you’d like to ignore.

See also https://github.com/errata-ai/vale/issues/141#issuecomment-519128843.

---

I know this may be inconvenient, but Vale has support for quite a few different formats. And, in order to avoid having to parse each one ourselves, we convert them all to HTML (using third-party libraries) before doing any linting.

So while it's possible to have "non-standard" blocks in Markdown (and other formats that are then converted to valid HTML), we don't support these features in HTML itself -- it's either a valid HTML file or it's not.

If you find that you need to lint content in files that are not strictly valid HTML, I'd suggest running Vale *after* you build your site with Hugo.

(Obviously, there are some drawbacks to this but it's hard to offer perfect support for every static site generator's take on templating and metadata).
